Automattic/jetpack
# Backup: Hide backup segments/info when Jetpack is connected but site has no backup plan

After some explorations, we noticed that there are some specific scenarios where backup segments/info like "Your site's heartbeat" were appearing in the pricing page of the backup plugin. It is occurring because when the site is connected to Jetpack, it is always assuming the site has a backup plan. A quick fix is required to receive the necessary variables as props instead of function arguments. 

## Proposed changes:

* Switch the BackupSegment component arguments to receive props instead of function arguments.   

### Screenshots
Before the change, the section that includes the paragraphs "Your site's heartbeat" and "Restore points created with every edit" are visible when the site has no backups but is connected to Jetpack. After the change, this section is no longer visible.

| Before | After |
|---|---|
| ![image](https://user-images.githubusercontent.com/1488641/221693483-83f804db-474b-4c26-bcc6-8b808c15ad82.png) | ![image](https://user-images.githubusercontent.com/1488641/221693581-0da75caa-ec37-4121-92ab-f99b534feaae.png) |

### Other information:


## Jetpack product discussion


p1675725651770819/1675715679.719879-slack-C025MP1Q3HT

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Spin up a JN site with Jetpack VaultPress Backup plugin using this branch. Don't include a backup plan.
* Navigate to Jetpack VaultPress Backup admin page and ensure the section that includes paragraphs like "Your site's heartbeat" and "Restore points created with every edit" are not visible.
* Click on `Get VaultPress Backup` button, connect the site to Jetpack but do not purchase a backup plan.
* Go back to the backup admin page and ensure the section is still not visible.
* Now purchase a plan, and after that, ensure the section is now visible.
